credhelper/helper.go

```diff
@@ -165,25 +160,28 @@ It looks for credentials in the following places, preferring the first location
     (In this final case any provided scopes are ignored.)
 */
 func tokenFromEnv() (string, error) {
-	ts, err := google.DefaultTokenSource(config.OAuthHTTPContext, config.GCRScopes...)
+	creds, err := cloudcreds.DetectDefault(&cloudcreds.DetectOptions{
+		Scopes:           config.GCRScopes,
+		UseSelfSignedJWT: true,
+	})
 	if err != nil {
-		return "", err
+		return "", helperErr("failed to detect default credentials", err)
 	}
 
-	token, err := ts.Token()
+	token, err := creds.Token(config.OAuthHTTPContext)
```

Comment:
nit: if you are not trying to do anything too fancy here, I would pass context.Background(). If you are trying to pass in a customer HTTP client, please use the configuration on the DetectOptions struct above.